### ISSUE TITLE
Compiler warning about babel-preset-es2015@6.24.1

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger-client-node#154]
happens during `npm install`

```
npm WARN deprecated babel-preset-es2015@6.24.1: We're super 😸  excited that you're trying to
use ES2015 syntax, but instead of continuing yearly presets 😭 , we recommend using 
babel-preset-env: npm install babel-preset-env. preset-env without options will compile ES2015+
down to ES5. And by targeting specific browsers, Babel can do less work and you can ship native 
ES2015+ to users 😎 ! Also, we are in the process of releasing v7, so give 
http://babeljs.io/blog/2017/09/12/planning-for-7.0 a read and test it! Thanks so much for using 
Babel 🙏 , please give us a follow @babeljs for updates, join slack.babeljs.io for discussion and help 
support at opencollective.com/babel
```

**Comment by MarckK** [comment]
For transpiling ES2015+ to ES5, Babel recommends using [babel-preset-env.](https://babeljs.io/env/) Without any configuration options, babel-preset-env behaves exactly the same as babel-preset-latest (or babel-preset-es2015, babel-preset-es2016, and babel-preset-es2017 together). You can also configure it to target certain browsers or runtime environments, and this will make your bundles smaller.

However, there is an issue with using preset-env and the [transform-class-properties](https://github.com/babel/babel-preset-env/issues/417) plugin that we are using.

**Comment by yurishkuro** [comment]
cc @tiffon

**Comment by tiffon** [comment]
@MarckK Thanks for your contributions!

The issue you referenced looks pretty gnarly, but things seem to work fine with the following changes:

```diff
diff --git a/.babelrc b/.babelrc
index 2e2fb35..d6795c9 100644
--- a/.babelrc
+++ b/.babelrc
@@ -1,4 +1,16 @@
 {
-  "presets": ["es2015"],
-  "plugins": ["transform-flow-strip-types", "transform-class-properties"]
-}
+  "presets": [
+    [
+      "env",
+      {
+        "targets": {
+          "node": "0.10"
+        }
+      }
+    ]
+  ],
+  "plugins": [
+    "transform-flow-strip-types",
+    "transform-class-properties"
+  ]
+}
\ No newline at end of file
diff --git a/Makefile b/Makefile
index 83a5271..0681331 100644
--- a/Makefile
+++ b/Makefile
@@ -17,9 +17,9 @@ test: build-node
 .PHONY: build-node
 build-node: node_modules
 	rm -rf ./dist/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/src/ src/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/test/ test/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/crossdock/ crossdock/
+	node_modules/.bin/babel --source-maps -d dist/src/ src/
+	node_modules/.bin/babel --source-maps -d dist/test/ test/
+	node_modules/.bin/babel --source-maps -d dist/crossdock/ crossdock/
 	cp -R ./test/thrift ./dist/test/thrift/
 	cp package.json ./dist/
 	cp -R ./src/jaeger-idl ./dist/src/
diff --git a/package.json b/package.json
index 1022901..aeb28b0 100644
--- a/package.json
+++ b/package.json
@@ -16,15 +16,16 @@
   ],
   "dependencies": {
     "node-int64": "^0.4.0",
+    "opentracing": "^0.13.0",
     "thriftrw": "^3.5.0",
-    "xorshift": "^0.2.0",
-    "opentracing": "^0.13.0"
+    "xorshift": "^0.2.0"
   },
   "devDependencies": {
     "babel-cli": "^6.11.4",
     "babel-istanbul": "^0.11.0",
     "babel-plugin-transform-class-properties": "^6.11.5",
     "babel-plugin-transform-flow-strip-types": "^6.8.0",
+    "babel-preset-env": "^1.6.1",
     "babel-preset-es2015": "^6.13.2",
     "babel-register": "^6.11.6",
     "beautify-benchmark": "^0.2.4",
```

I also noticed [create-react-app](https://github.com/facebookincubator/create-react-app/blob/master/packages/babel-preset-react-app/index.js#L11) is using it without issue (maybe because they're [not using](https://github.com/babel/babel-preset-env/issues/417#issuecomment-333395406) `.babel#env`?).

Side note: I did run into an unrelated issue with flow, though: https://github.com/facebook/flow/issues/4146

**Comment by MarckK** [comment]
Thank you, @tiffon! I didn't have an issue with flow, re the issue mentioned. I had thought perhaps it had to do with the ordering of the plugins, but I'm not getting that error, with `transform-class-properties` ordered first or second. If it's still an issue, I'd need more information.

On the other hand, I am having a Travis failure, with error: `Error: Cannot find module 'process' (While processing preset: "/home/travis/build/jaegertracing/jaeger-client-node/node_modules/babel-preset-env/lib/index.js")`

**Comment by MarckK** [comment]
Hmm, the build errors I'm getting seemed to have been a [problem with v1.3.0](https://github.com/babel/babel-preset-env/issues/235), but a smoke test was added to babel-preset-env to fix the issue (and we are using v1.6.1).
